src/models/ContractEntity.ts

```diff
@@ -1,13 +1,14 @@
 import BigNumber from 'bignumber.js';
 import Comparable from '../observer/Comparable';
+import { EntityType } from '../repositories/ContractEntityRepository';
```

Comment:
This introduces cyclic dependency: ContractEntity includes ContractEntityRepository and vice versa.
Please, move EntityType enum into ContractEntity.ts, this will eliminate cyclic dependency and will bring type closer to model.